internal/pkg/cli/app_show_test.go

```diff
@@ -142,6 +179,74 @@ func TestAppShow_Ask(t *testing.T) {
 				m.EXPECT().SelectOne(applicationShowProjectNamePrompt, applicationShowProjectNameHelpPrompt, []string{"my-project", "archer-project"}).Return("my-project", nil).Times(1)
 				m.EXPECT().SelectOne(fmt.Sprintf(applicationShowAppNamePrompt, "my-project"), applicationShowAppNameHelpPrompt, []string{"my-app", "archer-app"}).Return("my-app", nil).Times(1)
 			},
+
+			mockWorkspace: func(m *mocks.MockWorkspace) {
+				m.EXPECT().Apps().Return([]archer.Manifest{}, nil)
+			},
+
+			wantedProject: "my-project",
+			wantedApp:     "my-app",
+			wantedError:   nil,
+		},
+		"retrieve local app names": {
+			inputProject: "",
+			inputApp:     "",
+
+			mockStoreReader: func(m *climocks.MockstoreReader) {
+				m.EXPECT().ListProjects().Return([]*archer.Project{
+					{Name: "my-project"},
+					{Name: "archer-project"},
+				}, nil)
+			},
+
+			mockPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne(applicationShowProjectNamePrompt, applicationShowProjectNameHelpPrompt, []string{"my-project", "archer-project"}).Return("my-project", nil).Times(1)
+				m.EXPECT().SelectOne(fmt.Sprintf(applicationShowAppNamePrompt, "my-project"), applicationShowAppNameHelpPrompt, []string{"my-app", "archer-app"}).Return("my-app", nil).Times(1)
+			},
+
+			mockWorkspace: func(m *mocks.MockWorkspace) {
+				m.EXPECT().Apps().Return([]archer.Manifest{
+					&manifest.LBFargateManifest{
+						AppManifest: manifest.AppManifest{
+							Name: "my-app",
+						},
+					},
+					&manifest.LBFargateManifest{
+						AppManifest: manifest.AppManifest{
+							Name: "archer-app",
+						},
+					},
+				}, nil)
+			},
+
+			wantedProject: "my-project",
+			wantedApp:     "my-app",
+			wantedError:   nil,
+		},
+		"skip selecting if only one application found": {
+			inputProject: "",
+			inputApp:     "",
+
+			mockStoreReader: func(m *climocks.MockstoreReader) {
+				m.EXPECT().ListProjects().Return([]*archer.Project{
+					{Name: "my-project"},
+					{Name: "archer-project"},
+				}, nil)
+				m.EXPECT().ListApplications("my-project").Return([]*archer.Application{
+					{
+						Name: "my-app",
+					},
+				}, nil)
+			},
+
+			mockPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne(applicationShowProjectNamePrompt, applicationShowProjectNameHelpPrompt, []string{"my-project", "archer-project"}).Return("my-project", nil).Times(1)
```

Comment:
should this be called at all if app selection should be skipped?

Reply:
This applies to the circumstance when the user is outside of the workspace and have more than one project. So the user needs to select the project you'd like to use before selecting app. However, since the user chooses "my-project" and there's only one app under that project so we don't prompt to choose app.

Reply:
Basically this is called for project selection instead of app selection...But to avoid misunderstanding I'd better to make this example in a workspace so that don't need to select project.